src/lib/InstantSearch.ts

```diff
@@ -263,14 +266,14 @@ See ${createDocumentationLink({
 
   /**
    * Hooks a middleware into the InstantSearch lifecycle.
-   *
-   * This method is considered as experimental and is subject to change in
-   * minor versions.
    */
   public use(...middleware: Middleware[]): this {
     const newMiddlewareList = middleware.map(fn => {
       const newMiddleware = fn({ instantSearchInstance: this });
-      this.middleware.push(newMiddleware);
+      this.middleware.push({
+        middleware: fn,
+        middlewareDefinition: newMiddleware,
+      });
```

Comment:
```suggestion
      this.middleware.push([fn,newMiddleware]);
```

this will prevent the name `middlewareDefinition` in the output. Will only have a tiny bundlesize impact though, so not sure if worth it.

Reply:
I tried it and `68.06` → `68.05`, so 0.01kb diff. 

```js
  public unuse(...middlewareToUnuse: Middleware[]): this {
    this.middleware
      .filter(m => middlewareToUnuse.includes(m[0]))
      .forEach(m => m[1].unsubscribe());

    this.middleware = this.middleware.filter(
      m => !middlewareToUnuse.includes(m[0])
    );

    return this;
  }
```

it looks a little bit confusing with [0] and [1].

We can use different names for the object instead of `middleware & middlewareDefinition` if you have any idea, like `fn & middleware`?

Reply:
what about `creator` and `instance`?

Reply:
since it's internal, no big deal, but let's go with creator